File: fortran/euler1.f
! Euler1 in Fortran90
 
program euler1
    integer :: euler

    print*, euler(999)
end

function euler(size)
    integer euler, size
    euler = 0

    do i = 0, size
        if (modulo(i,3)==0 .or. modulo(i,5)==0) then
            euler = euler + i
        end if
    end do
end

File: haskell/euler1.hs
-- Euler1 in Haskell

mysum :: Integer -> Integer -> Integer
mysum mul lim = foldr (+) 0 [0, mul .. lim-1]

euler1 :: Integer -> Integer
euler1 (n) = mysum 3 n + mysum 5 n - mysum 15 n

main :: IO ()
main = do
    let result = euler1 1000 
    putStrLn $ "euler1 = " ++ show result 

